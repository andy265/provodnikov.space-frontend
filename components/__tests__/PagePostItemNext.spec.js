const { createRenderer } = require('vue-server-renderer')
const { createLocalVue, shallow } = require('vue-test-utils')
jest.mock('~/components/HasModelMixin', () => require('HasModelMixin'))
const PagePostItemNext = require('~/components/PagePostItemNext')

Object.defineProperty(global.location, 'href', {
  writable: true
})

describe('PagePostItemNext.vue', () => {
  let localVue
  let mocks
  let propsData
  let options

  beforeEach(() => {
    localVue = createLocalVue()
    mocks = {
      $router: {
        replace: jest.fn()
      },
      $route: {
        path: '/post/some_post',
        hash: ''
      },
      $scrollTo: jest.fn()
    }
    propsData = {
      state: 'index',
      isPreview: true,
      postId: 'some_post',
      postUrl: 'http://localhost:3000/post/some_post',
      value: {}
    }
    options = {
      localVue,
      mocks,
      propsData
    }
  })

  it('has same HTML structure when state is "index" and it is preview', () => {
    let Component = Object.assign({}, PagePostItemNext, { components: {} })
    let wrapper = shallow(Component, options)
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('has same HTML structure when state is "index" and it isn`t preview', () => {
    options.propsData.isPreview = false
    let Component = Object.assign({}, PagePostItemNext, { components: {} })
    let wrapper = shallow(Component, options)
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('scroll page to its element and remove "#next" from path if (state is "index") and (it isn`t preview) and (hash is "#next")', () => {
    options.propsData.isPreview = false
    mocks.$route.hash = '#next'
    let wrapper = shallow(PagePostItemNext, options)
    expect(wrapper.vm.$router.replace.mock.calls[0][0]).toBe(wrapper.vm.$route.path)
    expect(wrapper.vm.$scrollTo.mock.calls).toHaveLength(1)
  })

  it('change route on "next" button click', () => {
    let wrapper = shallow(PagePostItemNext, options)
    wrapper.vm.onNextClick()
    expect(global.location.href).toBe('http://localhost:3000/post/some_post#next')
  })
})
