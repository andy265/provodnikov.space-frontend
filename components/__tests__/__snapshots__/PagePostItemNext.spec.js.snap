// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePostItemNext.vue has same HTML structure when state is "index" and it is preview 1`] = `
<div class="page-post-item-next">
    <AppButton class="app-button_base">дальше
        <AppIcon type="long-arrow-right" class="app-icon_theme_button app-icon_middle"></AppIcon>
    </AppButton>
    <!---->
    <!---->
</div>
`;

exports[`PagePostItemNext.vue has same HTML structure when state is "index" and it isn\`t preview 1`] = `
<div class="page-post-item-next">
    <!---->
    <div class="page-post-item-next__padding"></div>
    <!---->
</div>
`;
